android/java/org/chromium/chrome/browser/crypto_wallet/activities/NetworkSelectorActivity.java

```diff
@@ -58,27 +72,48 @@ public void finishNativeInitialization() {
         initState();
     }
 
+    @SuppressLint("NotifyDataSetChanged")
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+        if (resultCode != RESULT_OK) return;
+
+        BraveActivity activity = BraveActivity.getBraveActivity();
+        if (activity != null) {
+            activity.getWalletModel().getCryptoModel().getNetworkModel().init();
+        }
+        networkSelectorAdapter.notifyDataSetChanged();
+    }
+
     private void initState() {
         BraveActivity activity = BraveActivity.getBraveActivity();
         if (activity != null) {
             mWalletModel = activity.getWalletModel();
+            activity.getWalletModel().getCryptoModel().getNetworkModel().init();
```

Comment:
is it required now?

Reply:
It covers the use case when a user adds/removes a network from `Settings` -> `Brave Wallet` -> Networks.
Without this line the changes were not updated.

Reply:
So I think it would be better to call `refreshnetworks` after `add network` action

Like, create an add network method in the `NetworkModel` (e.g. `setNetwork `) and in case of success, trigger the `refreshnetwork or init` then, we won't have to manually update the network list

Reply:
or can simply call `refreshnetwork or init` after `add network` success for quick implementation.

Reply:
Moved `refreshNetworks` method call in `BraveWalletNetworksPreference#updateNetworksList()`. 

`updateNetworksList()` is called every time a network is removed, or modified.

Reply:
nice, I think we can now remove [onActivityResult](https://github.com/brave/brave-core/blob/aa2ec267d407badd0bd1bb7ce53aff15b37dbd83/android/java/org/chromium/chrome/browser/crypto_wallet/activities/NetworkSelectorActivity.java#L81) and startactivityforresult. Proceed accordingly.

Reply:
Done in [commit `98fdf90`](https://github.com/brave/brave-core/pull/16481/commits/98fdf90c957239577388c83ba858448bde084ffd)

I'm merging the PR as soon as all the checks are green.